fix(portal): Use modal for confirmation dialogs

AndrewDryga opened the pull request:
Still need to replace a few more `data-confirm` actions (have an issue with HTML <dialog> positioning when it's defined inside Tailwind tables.)

Closes https://github.com/firezone/firezone/issues/5794
Closes #5766
Closes #5887

AndrewDryga committed:
Update wording for Activities

Closes #5794

AndrewDryga committed:
Replace all alerts with modal dialogs for Danger Zone actions

jamilbk commented on elixir/apps/web/assets/js/hooks.js:
```diff
@@ -94,4 +94,22 @@ Hooks.Copy = {
   },
 };
 
+Hooks.ConfirmDialog = {
```
```suggestion
Hooks.CloseConfirmDialog = {
```

AndrewDryga replied:
This hook doesn't close the dialog but opens it.

jamilbk commented on elixir/apps/web/lib/web/components/form_components.ex:
```diff
@@ -358,6 +358,85 @@ defmodule Web.FormComponents do
+    <dialog
+      id={"#{@id}_dialog"}
+      class={[
+        "backdrop:bg-slate-800/75 bg-transparent",
```
```suggestion
        "backdrop:bg-gray-800/75 bg-transparent",
```

jamilbk commented on elixir/apps/web/lib/web/components/form_components.ex:
```diff
@@ -358,6 +358,85 @@ defmodule Web.FormComponents do
+        </div>
+      </form>
+    </dialog>
+    <.button id={@id} style={@style} size={@size} icon={@icon} class={@class} phx-hook="ConfirmDialog">
```
```suggestion
    <.button id={@id} style={@style} size={@size} icon={@icon} class={@class} phx-hook="CloseConfirmDialog">
```

jamilbk commented on elixir/apps/web/lib/web/components/form_components.ex:
```diff
@@ -358,6 +358,85 @@ defmodule Web.FormComponents do
+      ]}
+    >
+      <form method="dialog">
+        <div class="relative bg-white rounded-lg shadow">
```
Closes on click outside 

```suggestion
        <div
          phx-click-away={JS.dispatch("click", to: "[data-dialog-action='cancel']")}
          class="relative bg-white rounded-lg shadow"
        >

```

jamilbk commented on elixir/apps/web/lib/web/live/actors/show.ex:
```diff
@@ -405,9 +405,9 @@ defmodule Web.Actors.Show do
+      <:title>Tunnel Sessions</:title>
       <:help>
-        Authorized attempts by actors to access the resource governed by this policy.
+        Authorized attempts by the actor to access a resource.
```
```suggestion
        Authorized sessions opened by this Actor to access a Resource.
```

jamilbk commented on elixir/apps/web/lib/web/live/actors/show.ex:
```diff
@@ -405,9 +405,9 @@ defmodule Web.Actors.Show do
 
     <.section>
-      <:title>Authorized Activity</:title>
+      <:title>Tunnel Sessions</:title>
```
```suggestion
      <:title>Authorized Sessions</:title>
```

jamilbk commented on elixir/apps/web/lib/web/live/clients/show.ex:
```diff
@@ -156,9 +156,9 @@ defmodule Web.Clients.Show do
 
     <.section>
-      <:title>Authorized Activity</:title>
+      <:title>Tunnel Sessions</:title>
```
```suggestion
      <:title>Authorized Sessions</:title>
```

jamilbk commented on elixir/apps/web/lib/web/live/clients/show.ex:
```diff
@@ -156,9 +156,9 @@ defmodule Web.Clients.Show do
+      <:title>Tunnel Sessions</:title>
       <:help>
-        Authorized attempts by actors to access the resource governed by this policy.
+        Authorized attempts by the client to access a resource.
```
```suggestion
        Authorized sessions opened by this Client to access a Resource.
```

jamilbk commented on elixir/apps/web/lib/web/live/gateways/show.ex:
```diff
@@ -108,12 +108,37 @@ defmodule Web.Gateways.Show do
+          <:dialog_title>Delete Gateway</:dialog_title>
+          <:dialog_content>
+            <p>
+              Are you sure you want to delete this gateway?
```
```suggestion
              Are you sure you want to delete this Gateway?
```

jamilbk commented on elixir/apps/web/lib/web/live/gateways/show.ex:
```diff
@@ -108,12 +108,37 @@ defmodule Web.Gateways.Show do
+              Are you sure you want to delete this gateway?
+            </p>
+            <p class="mt-4 text-sm">
+              Deleting the gateway does not remove it's access token so it can be re-created again,
+              revoke the token on the
+              <.link
+                navigate={~p"/#{@account}/sites/#{@gateway.group}"}
+                class={["font-medium", link_style()]}
+              >
+                site
+              </.link>
+              page if you want to prevent the gateway from connecting to the portal.
```
```suggestion
              <strong>Note:</strong>
              Deleting a Gateway does not remove its access token.
              <.link
                navigate={~p"/#{@account}/sites/#{@gateway.group}"}
                class={["font-medium", link_style()]}
              >
                Revoke
              </.link>
              the Gateway token instead to prevent it from being re-created
              automatically the next time it starts.
```

jamilbk commented on elixir/apps/web/lib/web/live/groups/show.ex:
```diff
@@ -240,12 +240,24 @@ defmodule Web.Groups.Show do
         >
+          <:dialog_title>Delete Group</:dialog_title>
+          <:dialog_content>
+            Are you sure you want to delete this group and all related policies?
```
```suggestion
            Are you sure you want to delete this Group and all related Policies?
```

jamilbk commented on elixir/apps/web/lib/web/live/policies/show.ex:
```diff
@@ -158,9 +158,9 @@ defmodule Web.Policies.Show do
 
     <.section>
-      <:title>Authorized Activity</:title>
+      <:title>Tunnel Sessions</:title>
```
```suggestion
      <:title>Authorized Sessions</:title>
```

jamilbk commented on elixir/apps/web/lib/web/live/policies/show.ex:
```diff
@@ -158,9 +158,9 @@ defmodule Web.Policies.Show do
+      <:title>Tunnel Sessions</:title>
       <:help>
-        Authorized attempts by actors to access the resource governed by this policy.
+        Authorized attempts by an actors to access the resource governed by this policy.
```
```suggestion
        Authorized sessions opened by Actors to access the Resources governed by this Policy.
```

jamilbk commented on elixir/apps/web/lib/web/live/policies/show.ex:
```diff
@@ -209,13 +209,25 @@ defmodule Web.Policies.Show do
         >
+          <:dialog_title>Delete Policy</:dialog_title>
+          <:dialog_content>
+            Are you sure you want to delete this Policy? All tunnel sessions authorized by it will be expired.
```
```suggestion
            Are you sure you want to delete this Policy? All sessions authorized by it will be expired.
```

jamilbk commented on elixir/apps/web/lib/web/live/resources/show.ex:
```diff
@@ -248,9 +248,9 @@ defmodule Web.Resources.Show do
 
     <.section>
-      <:title>Authorized Activity</:title>
+      <:title>Tunnel Sessions</:title>
```
```suggestion
      <:title>Authorized Sessions</:title>
```

jamilbk commented on elixir/apps/web/lib/web/live/resources/show.ex:
```diff
@@ -248,9 +248,9 @@ defmodule Web.Resources.Show do
+      <:title>Tunnel Sessions</:title>
       <:help>
-        Authorized attempts by actors to access the resource governed by this policy.
+        Authorized attempts by actors to access the resource.
```
```suggestion
        Authorized sessions opened by Actors to access this Resource.
```

jamilbk commented on elixir/apps/web/lib/web/live/resources/show.ex:
```diff
@@ -304,13 +304,25 @@ defmodule Web.Resources.Show do
         >
+          <:dialog_title>Delete Resource</:dialog_title>
+          <:dialog_content>
+            Are you sure want to delete this resource along with all associated policies and tunnel sessions?
```
```suggestion
            Are you sure want to delete this Resource along with all associated Policies?
            This will immediately end all active sessions opened for this Resource.
```

jamilbk commented on elixir/apps/web/lib/web/live/settings/identity_providers/google_workspace/show.ex:
```diff
@@ -155,12 +155,24 @@ defmodule Web.Settings.IdentityProviders.GoogleWorkspace.Show do
         >
+          <:dialog_title>Delete Identity Provider</:dialog_title>
+          <:dialog_content>
+            Are you sure want to delete this provider along with all related data?
```
```suggestion
            Are you sure want to delete this provider? This will remove
            <strong>all</strong> Actors and Groups associated with this provider.
```

jamilbk commented on elixir/apps/web/lib/web/live/settings/identity_providers/jumpcloud/show.ex:
```diff
@@ -167,12 +167,24 @@ defmodule Web.Settings.IdentityProviders.JumpCloud.Show do
         >
+          <:dialog_title>Delete Identity Provider</:dialog_title>
+          <:dialog_content>
+            Are you sure want to delete this provider along with all related data?
```
```suggestion
            Are you sure want to delete this provider? This will remove
            <strong>all</strong> Actors and Groups associated with this provider.
```

jamilbk commented on elixir/apps/web/lib/web/live/settings/identity_providers/microsoft_entra/show.ex:
```diff
@@ -155,12 +155,24 @@ defmodule Web.Settings.IdentityProviders.MicrosoftEntra.Show do
         >
+          <:dialog_title>Delete Identity Provider</:dialog_title>
+          <:dialog_content>
+            Are you sure want to delete this provider along with all related data?
```
```suggestion
            Are you sure want to delete this provider? This will remove
            <strong>all</strong> Actors and Groups associated with this provider.
```

jamilbk commented on elixir/apps/web/lib/web/live/settings/identity_providers/okta/show.ex:
```diff
@@ -173,12 +173,24 @@ defmodule Web.Settings.IdentityProviders.Okta.Show do
         >
+          <:dialog_title>Delete Identity Provider</:dialog_title>
+          <:dialog_content>
+            Are you sure want to delete this provider along with all related data?
```
```suggestion
            Are you sure want to delete this provider? This will remove
            <strong>all</strong> Actors and Groups associated with this provider.
```

jamilbk commented on elixir/apps/web/lib/web/live/settings/identity_providers/openid_connect/show.ex:
```diff
@@ -134,12 +134,24 @@ defmodule Web.Settings.IdentityProviders.OpenIDConnect.Show do
         >
+          <:dialog_title>Delete Identity Provider</:dialog_title>
+          <:dialog_content>
+            Are you sure want to delete this provider along with all related data?
```
```suggestion
            Are you sure want to delete this provider? This will remove
            <strong>all</strong> Actors and Groups associated with this provider.
```

jamilbk commented on elixir/apps/web/lib/web/live/sites/show.ex:
```diff
@@ -260,12 +260,24 @@ defmodule Web.Sites.Show do
         >
+          <:dialog_title>Delete Site</:dialog_title>
+          <:dialog_content>
+            Are you sure you want to delete this Site and disconnect all its Gateways?
```
```suggestion
            Are you sure you want to delete this Site? This will
            <strong>immediately</strong> disconnect all associated Gateways.
```

jamilbk reviewed (changes requested):
Looks great, thanks for taking this on!

I made a few UX and spelling/grammar suggestions. One of them is to settle on the `Authorized sessions` terminology instead of `Tunnel sessions`.

I did some thinking about it this morning and we can build off that terminology to describe how the Policy Engine works. We can also leverage the "session" terminology within the context of Policies when we add re-authentication periods.

AndrewDryga committed:
Apply content suggestions from code review

Co-authored-by: Jamil <jamilbk@users.noreply.github.com>
Signed-off-by: Andrew Dryga <andrew@dryga.com>

ReactorScram committed:
fix(portal): "Are you sure want" --> "Are you sure you want" (#5888)

Closes #5887

Nit: We could say just "Enable this policy?" and reduce the word count
to less than half. It's not perfect by-the-book English but then users
can scan the dialog without thinking, "Oh no, an entire sentence."

I think many programs do that for confirmation dialogs.

AndrewDryga committed:
Fix modal appearance and replace rest of confirmation dialogs

AndrewDryga committed:
Fix tests

jamilbk reviewed (approved):
LGTM
